board/netgear/readynas_duo_v2/readynas_duo_v2.c

```diff
@@ -0,0 +1,301 @@
+		0
+	};
+	kirkwood_mpp_conf(kwmpp_config, NULL);
+	return 0;
+}
+
+int board_init(void)
+{
+	char **val1, **val2; // variables for getting the 'yes' or 'no' switch from some environment variables
+	val1="";
+	val2="";
+	/*
+	 * arch number of board
+	 */
+	gd->bd->bi_arch_number = MACH_TYPE_DB88F6281_BP;
+
+	/* address of boot parameters */
+	gd->bd->bi_boot_params = mvebu_sdram_bar(0) + 0x100;
+
+    kw_gpio_set_valid(22, 1); /* pmx-led-blue-disk2 */
+    kw_gpio_set_valid(23, 1); /* pmx-led-blue-disk1 */
+    kw_gpio_set_valid(26, 1); /* HDD0 Power */
+    kw_gpio_set_valid(28, 1); /* HDD1 power */
+    kw_gpio_set_valid(29, 1); /* pmx-led-blue-backup */
+    kw_gpio_set_valid(31, 1); /* pmx-led-blue-power */
+    kw_gpio_set_valid(35, 1); /* USB 2.0 Power */
+    kw_gpio_set_valid(38, 1); /* pmx-led-blue-activity */
+    kw_gpio_set_valid(46, 1); /* USB 3.0 Power */
+
+    // Enable power LED
+    kw_gpio_direction_output(31, 0);
+	kw_gpio_set_value(31, 0);
+
+	// initialize the cooling fan
+	unsigned char byte=0x40; // the higher the hex value, the faster the fan
+	int fi;
+	// try fan init 10 times, due to unreliable i2c access
+	for (fi = 0; fi < 10; fi++)
+	{
+		if(i2c_write(CONFIG_SYS_I2C_G762_ADDR, 0x0, 1, &byte, 1) != 0)
+		{
+			if(fi == 9)
+			{
+				puts ("Error writing the i2c chip : G76x(Fan controller).\n");
+			}
+			else
+			{
+				continue;
+			}
+			udelay(50000);
+		}
+		else
+		{
+			puts ("G762 Fan controller: Successfully initialized.\n");
+			break;
+		}
+	}
+
+	// Enabling USB 2.0 Power
+	puts ("Enabled USB 2.0 Power.\n");
+	kw_gpio_direction_output(35, 0);
+	kw_gpio_set_value(35, 1);
+	val1="";
+
+	// Enabling USB 3.0 Power
+	puts ("Enabled USB 3.0 Power.\n");
+	kw_gpio_direction_output(46, 0);
+	kw_gpio_set_value(46, 1);
+
+	puts ("Running HDD staggered spin-up...\n"); // be a little easier on the power supply
+	puts ("Enabling HDD-0 Power in 1s.\n");
+	udelay(1000000);
+	kw_gpio_direction_output(26, 0);
+	kw_gpio_set_value(26, 1);
+	puts ("Enabling HDD-1 Power in 3s.\n");
+	udelay(3000000);
+	kw_gpio_direction_output(28, 0);
+	kw_gpio_set_value(28, 1);
+
+	return 0;
+}
+
+
+#ifdef CONFIG_RESET_PHY_R
+/* Configure and enable MV88E1318 PHY */
+void reset_phy(void)
+{
+	u16 reg;
+	u16 devadr;
+	char *name = "egiga0";
+
+	if (miiphy_set_current_dev(name))
+		return;
+
+	/* command to read PHY dev address */
+	if (miiphy_read(name, 0xEE, 0xEE, (u16 *) &devadr)) {
+		printf("Err..%s could not read PHY dev address\n",
+			__FUNCTION__);
+		return;
+	}
+
+	/* Set RGMII delay */
+	miiphy_write(name, devadr, MV88E1318_PGADR_REG, MV88E1318_MAC_CTRL_PG);
+	miiphy_read(name, devadr, MV88E1318_MAC_CTRL_REG, &reg);
+	reg |= (MV88E1318_RGMII_RXTM_CTRL | MV88E1318_RGMII_TXTM_CTRL);
+	miiphy_write(name, devadr, MV88E1318_MAC_CTRL_REG, reg);
+	miiphy_write(name, devadr, MV88E1318_PGADR_REG, 0);
+
+	// commented out PHY reset, as it always failed, anyway (worked around via mii tool call in 'preboot' variable)
+	/* reset the phy */
```

Comment:
This Reset PHY function needs to be revised. As it is, the implementation in this function is specific to the NSA325, 310s, and 320. A typical and simpe PHY reset can be found in board/Seagate/dockstar. 

I'm wondering that probably why it caused error and you needed the work-around with mmi.

Reply:
Agreed. And thanks for the hint. Will look into it, when I find the time.